### PR TITLE
Add test for awesome window manager

### PR DESCRIPTION
**Pull request by asdil12**
Based on "minimalx" installation.

Related issue: https://progress.opensuse.org/issues/9522

### CODE REVIEW

File: tests/console/consoletest_finish.pm [hunk 1]
```diff
@@ -74,7 +74,14 @@ sub run() {
         }
         wait_idle;
         mouse_hide(1);
-        assert_screen 'generic-desktop';
+        # in case of awesome test we use xdm and need to log in at the display
+        # manager
+        if (check_var("DESKTOP", "awesome")) {
+            assert_screen 'displaymanager', 200;
+        }
```
Comment (coolo) [inline tests/console/consoletest_finish.pm]: see my comments in #930 - asserting here for displaymanager is wrong. At this point we *need* to be at the desktop

**Comment by coolo** [comment]
I'm against merging this hack - see my comments in #930 why I believe it's abusing $DESKTOP

**Comment by asdil12** [comment]
I forgot some changes - the new commit has the actual changes

**Comment by asdil12** [comment]
Resolved rebasing conflicts for new master

**Comment by asdil12** [comment]
@coolo OK to merge now?

**Comment by asdil12** [comment]
- Intruduced `DESKTOP_MINIMALX_INSTONLY` env var.
- Selected awesome using sysconfig
- Added *fatal* flag to the awesome tests.

File: tests/console/install_windowmanager.pm [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+use testapi;
+
+sub run() {
+    my $self = shift;
+    become_root();
+    if (check_var("DESKTOP", "awesome")) {
+        assert_script_run("zypper -n in awesome");
+        script_run("sed -i 's/^DEFAULT_WM.*\$/DEFAULT_WM=\"awesome\"/' /etc/sysconfig/windowmanager");
+    }
+    script_run("exit");
```
Comment (coolo) [inline tests/console/install_windowmanager.pm]: as I'm just replacing it - can you please avoid using script_run for exit - it's not a script, so just type_string "exit\n" please

(and yes, that's a pretty recent requirement)

**Comment by coolo** [comment]
if you don't get around it, I'll merge and fix it in my own branch

**Comment by okurz** [comment]
@coolo I assume @asdil12 can work on it on monday afternoon but feel free to adapt and merge in before

**Comment by asdil12** [comment]
@coolo Replaced with `type_string`

**Comment by coolo** [comment]
awesome is scheduled for next TW snapshot. Keep an eye on it

**Pull request merged**